miden-lib/asm/sat/internal/main.masm

```diff
@@ -74,6 +74,10 @@ export.main
     exec.layout::get_total_num_consumed_notes
     # => [num_consumed_notes]
 
+    # compute the pointer at which consumed note looping should terminate
```

Comment:
minor nit: I'd probably make the comment a bit more explicit. For example: "compute the pointer to the last consumed note (i.e., the pointer after which the looping should terminate)".

Reply:
I don't think this is quite accurate, it's not a pointer to the last consumed note.  It's a pointer to the note after the last consumed note.  I've gone with "compute the pointer to the consumed note after the last consumed note (i.e. the pointer at which the looping should terminate)"

Reply:
I still don't think this reads particularly well so please feel free to suggest alternatives.

Reply:
I think what you have now is fine (or at least I think it is more clear than the original version).